pkg/kudoctl/cmd/init/manager.go

```diff
@@ -0,0 +1,218 @@
+package init
+
+import (
+	"fmt"
+
+	"github.com/kudobuilder/kudo/pkg/version"
+
+	"k8s.io/api/apps/v1beta2"
+	v1 "k8s.io/api/core/v1"
+	apiextensionsclient "k8s.io/apiextensions-apiserver/pkg/client/clientset/clientset"
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/intstr"
+	"k8s.io/client-go/kubernetes"
+	clientv1beta2 "k8s.io/client-go/kubernetes/typed/apps/v1beta2"
+	corev1 "k8s.io/client-go/kubernetes/typed/core/v1"
+	"sigs.k8s.io/yaml"
+)
+
+//Defines the deployment of the KUDO manager and it's service definition.
+
+const (
+	group              = "kudo.dev"
+	crdVersion         = "v1alpha1"
+	defaultns          = "kudo-system"
+	defaultGracePeriod = 10
+)
+
+// Options is the configurable options to init
+type Options struct {
+	// Version is the version of the manager `0.5.0` for example (must NOT include the `v` in `v0.5.0`)
+	Version string
+	// Namespace to init into (default is kudo-system)
+	Namespace string
+	// TerminationGracePeriodSeconds defines the termination grace period for a pod
+	TerminationGracePeriodSeconds int64
+	// Image defines the image to be used
+	Image string
+}
+
+// NewOptions provides an option struct with defaults
+func NewOptions(v string) Options {
+
+	if v == "" {
+		v = version.Get().GitVersion
+	}
+
+	return Options{
+		Version:                       v,
+		Namespace:                     defaultns,
+		TerminationGracePeriodSeconds: defaultGracePeriod,
+		Image:                         fmt.Sprintf("kudobuilder/controller:v%v", v),
+	}
+}
+
+// Install uses Kubernetes client to install KUDO manager.
+func Install(client kubernetes.Interface, extClient *apiextensionsclient.Clientset, opts Options) error {
+	if err := installPrereqs(client, opts); err != nil {
+		return err
+	}
+	if err := installCrds(extClient); err != nil {
+		return err
+	}
+	if err := installManager(client, opts); err != nil {
+		return err
+	}
+	return nil
+}
+
+// Install uses Kubernetes client to install KUDO manager.
+func installManager(client kubernetes.Interface, opts Options) error {
+
+	if err := installStatefulSet(client.AppsV1beta2(), opts); err != nil {
+		return err
+	}
+
+	if err := installService(client.CoreV1(), opts); err != nil {
+		return err
+	}
+	return nil
+}
+
+func installStatefulSet(client clientv1beta2.StatefulSetsGetter, opts Options) error {
+	ss := generateDeployment(opts)
+	_, err := client.StatefulSets(opts.Namespace).Create(ss)
```

Comment:
I think all creates in this code will fail if the crds or something already exist. Should we really fail? Maybe it's fine and we should move on to another step?

Reply:
or maybe this should be another flag? if you agree, do you mind opening issue?